### ISSUE TITLE
Code reproduction error

### ISSUE THREAD

**Issue by ksk2023** [ClaudiaCuttano/SAMWISE#9]
Dear author, thank you very much for your contribution to the VOS field. I reproduced your paper as soon as possible. However, during the reproduction process following the steps you provided, I encountered issues like "Segmentation fault (core dumped)" and "Missing Keys". Below are my error messages. If you have time, I sincerely hope you could take a look. Thank you so much for your help! Wish you all the best!

Error Infomation:

Not using distributed mode
git:
  sha: ade600210b7d5c94da1ff5a9841d6c1d7d6ace89, status: has uncommited changes, branch: main

Namespace(lr=1e-05, batch_size=1, batch_size_val=1, num_frames=8, weight_decay=0, epochs=10, lr_drop=[60000], clip_max_norm=1, sam2_version='base', disable_pred_obj_score=False, motion_prompt=False, HSA=True, HSA_patch_size=[8, 4, 2], adapter_dim=256, fusion_stages_txt=[4, 8, 12], fusion_stages=[1, 2, 3], use_cme_head=True, switch_mem='reweight', cme_decision_window=4, dataset_file='ytvos', coco_path='data/coco', ytvos_path='data/ref-youtube-vos', davis_path='data/ref-davis', mevis_path='data/MeViS_release', max_size=1024, augm_resize=False, output_dir='output/first', name_exp='first', device='cuda', seed=0, resume='pretrain/pretrained_model.pth', resume_optimizer=False, start_epoch=0, eval=False, num_workers=4, no_distributed=False, threshold=0.5, split='valid', visualize=False, eval_clip_window=8, set='val', task='unsupervised', results_path=None, num_gpus=4, distributed=False, ngpu=1, log_file='output/first/log.txt')
number of params: 210766237
Trainable parameters:  4907458
Parameters fixed:  205858779
Total clips:  0

 video num:  3471  clip num:  49156

Missing Keys: ['sam.no_obj_ptr', 'sam.sam_mask_decoder.obj_score_token.weight', 'sam.sam_mask_decoder.pred_obj_score_head.layers.0.weight', 'sam.sam_mask_decoder.pred_obj_score_head.layers.0.bias', 'sam.sam_mask_decoder.pred_obj_score_head.layers.1.weight', 'sam.sam_mask_decoder.pred_obj_score_head.layers.1.bias', 'sam.sam_mask_decoder.pred_obj_score_head.layers.2.weight', 'sam.sam_mask_decoder.pred_obj_score_head.layers.2.bias', 'cmt_adapters.0.spatial_pos_embed', 'cmt_adapters.0.hsa.self_attn.in_proj_weight', 'cmt_adapters.0.hsa.self_attn.in_proj_bias', 'cmt_adapters.0.hsa.self_attn.out_proj.weight', 'cmt_adapters.0.hsa.self_attn.out_proj.bias', 'cmt_adapters.0.hsa.linear.weight', 'cmt_adapters.0.hsa.linear.bias', 'cmt_adapters.0.norm_pre_HSA.weight', 'cmt_adapters.0.norm_pre_HSA.bias', 'cmt_adapters.1.spatial_pos_embed', 'cmt_adapters.1.hsa.self_attn.in_proj_weight', 'cmt_adapters.1.hsa.self_attn.in_proj_bias', 'cmt_adapters.1.hsa.self_attn.out_proj.weight', 'cmt_adapters.1.hsa.self_attn.out_proj.bias', 'cmt_adapters.1.hsa.linear.weight', 'cmt_adapters.1.hsa.linear.bias', 'cmt_adapters.1.norm_pre_HSA.weight', 'cmt_adapters.1.norm_pre_HSA.bias', 'cmt_adapters.2.spatial_pos_embed', 'cmt_adapters.2.hsa.self_attn.in_proj_weight', 'cmt_adapters.2.hsa.self_attn.in_proj_bias', 'cmt_adapters.2.hsa.self_attn.out_proj.weight', 'cmt_adapters.2.hsa.self_attn.out_proj.bias', 'cmt_adapters.2.hsa.linear.weight', 'cmt_adapters.2.hsa.linear.bias', 'cmt_adapters.2.norm_pre_HSA.weight', 'cmt_adapters.2.norm_pre_HSA.bias']
Start training
段错误 (核心已转储)

**Comment by ga1i13o** [comment]
Closing the issue. Feel free to re-open if the issue persists or you have further questions

**Comment by ksk2023** [comment]
Dear author，Here's my hardware specs and environment details. I followed your guide exactly to set up my environment, but I'm still getting a "segmentation fault (core dumped)" error. I'm really stuck here - do you have any other troubleshooting ideas? Would you be willing to take a quick look at my setup? Alternatively, could you share your environment configuration with me as a reference? I'd really appreciate any help you can offer!



**Hardware information**

| NVIDIA-SMI 550.120                Driver Version: 550.120        CUDA Version: 12.4     |
|-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA A800 80GB PCIe          Off |   00000000:0A:00.0 Off |                    0 |
| N/A   49C    P0             57W /  300W |      17MiB /  81920MiB |      0%      Default |
|                                         |                        |             Disabled |
+-----------------------------------------+------------------------+----------------------+
|   1  NVIDIA A800 80GB PCIe          Off |   00000000:0B:00.0 Off |                    0 |
| N/A   50C    P0             57W /  300W |      17MiB /  81920MiB |      0%      Default |
|                                         |                        |             Disabled |
+-----------------------------------------+------------------------+----------------------+
|   2  NVIDIA A800 80GB PCIe          Off |   00000000:0E:00.0 Off |                    0 |
| N/A   59C    P0             86W /  300W |    4677MiB /  81920MiB |      0%      Default |
|                                         |                        |             Disabled |
+-----------------------------------------+------------------------+----------------------+
|   3  NVIDIA A800 80GB PCIe          Off |   00000000:12:00.0 Off |                    0 |
| N/A   48C    P0             57W /  300W |      17MiB /  81920MiB |      0%      Default |
|                                         |                        |             Disabled |
+-----------------------------------------+------------------------+----------------------+
|   4  NVIDIA A800 80GB PCIe          Off |   00000000:21:00.0 Off |                    0 |
| N/A   47C    P0             54W /  300W |      17MiB /  81920MiB |      0%      Default |
|                                         |                        |             Disabled |
+-----------------------------------------+------------------------+----------------------+
|   5  NVIDIA A800 80GB PCIe          Off |   00000000:22:00.0 Off |                    0 |
| N/A   49C    P0             59W /  300W |      17MiB /  81920MiB |      0%      Default |
|                                         |                        |             Disabled |
+-----------------------------------------+------------------------+----------------------+
|   6  NVIDIA A800 80GB PCIe          Off |   00000000:25:00.0 Off |                    0 |
| N/A   49C    P0             56W /  300W |      17MiB /  81920MiB |      0%      Default |
|                                         |                        |             Disabled |
+-----------------------------------------+------------------------+----------------------+
|   7  NVIDIA A800 80GB PCIe          Off |   00000000:29:00.0 Off |                    0 |
| N/A   47C    P0             56W /  300W |      17MiB /  81920MiB |      0%      Default |
|                                         |                        |             Disabled |
+-----------------------------------------+------------------------+----------------------+


**Dependencies information:**
  - _libgcc_mutex=0.1=conda_forge
  - _openmp_mutex=4.5=2_gnu
  - bzip2=1.0.8=h4bc722e_7
  - ca-certificates=2025.1.31=hbcca054_0
  - cudatoolkit=11.8.0=h4ba93d1_13
  - ld_impl_linux-64=2.43=h712a8e2_4
  - libffi=3.4.6=h2dba641_0
  - libgcc=14.2.0=h767d61c_2
  - libgcc-ng=14.2.0=h69a702a_2
  - libgomp=14.2.0=h767d61c_2
  - liblzma=5.6.4=hb9d3cd8_0
  - libnsl=2.0.1=hd590300_0
  - libsqlite=3.49.1=hee588c1_2
  - libstdcxx=14.2.0=h8f9b012_2
  - libstdcxx-ng=14.2.0=h4852527_2
  - libuuid=2.38.1=h0b41bf4_0
  - libxcrypt=4.4.36=hd590300_1
  - libzlib=1.3.1=hb9d3cd8_2
  - ncurses=6.5=h2d0b736_3
  - openssl=3.4.1=h7b32b05_0
  - pip=25.0.1=pyh8b19718_0
  - python=3.10.16=he725a3c_1_cpython
  - readline=8.2=h8c095d6_2
  - setuptools=75.8.2=pyhff2d567_0
  - tk=8.6.13=noxft_h4845f30_101
  - wheel=0.45.1=pyhd8ed1ab_1
  - pip:
      - annotated-types==0.7.0
      - antlr4-python3-runtime==4.9.3
      - beautifulsoup4==4.13.3
      - bitarray==3.2.0
      - blis==0.7.11
      - catalogue==2.0.10
      - certifi==2025.1.31
      - charset-normalizer==3.4.1
      - click==8.1.8
      - cloudpathlib==0.16.0
      - colorama==0.4.6
      - confection==0.1.5
      - contourpy==1.3.1
      - cycler==0.12.1
      - cymem==2.0.11
      - docker-pycreds==0.4.0
      - einops==0.8.0
      - filelock==3.13.1
      - fonttools==4.56.0
      - fsspec==2024.6.1
      - fvcore==0.1.5.post20221221
      - gdown==5.2.0
      - gitdb==4.0.12
      - gitpython==3.1.44
      - h5py==3.13.0
      - huggingface-hub==0.29.3
      - hydra-core==1.3.2
      - idna==3.10
      - imageio==2.37.0
      - imgaug==0.4.0
      - iopath==0.1.10
      - jinja2==3.1.4
      - joblib==1.4.2
      - kiwisolver==1.4.8
      - langcodes==3.5.0
      - language-data==1.3.0
      - lazy-loader==0.4
      - lxml==5.3.1
      - marisa-trie==1.2.1
      - markupsafe==2.1.5
      - matplotlib==3.10.1
      - mpmath==1.3.0
      - murmurhash==1.0.12
      - networkx==3.3
      - numpy==1.26.4
      - nvidia-cublas-cu11==11.11.3.6
      - nvidia-cuda-cupti-cu11==11.8.87
      - nvidia-cuda-nvrtc-cu11==11.8.89
      - nvidia-cuda-runtime-cu11==11.8.89
      - nvidia-cudnn-cu11==8.7.0.84
      - nvidia-cufft-cu11==10.9.0.58
      - nvidia-curand-cu11==10.3.0.86
      - nvidia-cusolver-cu11==11.4.1.48
      - nvidia-cusparse-cu11==11.7.5.86
      - nvidia-nccl-cu11==2.20.5
      - nvidia-nvtx-cu11==11.8.86
      - omegaconf==2.3.0
      - opencv-python==4.10.0.84
      - packaging==24.2
      - pandas==2.0.3
      - pillow==11.0.0
      - platformdirs==4.3.7
      - portalocker==3.1.1
      - preshed==3.0.9
      - protobuf==5.29.4
      - psutil==7.0.0
      - py3-wget==1.0.5
      - pyav==14.2.1
      - pycocotools==2.0.7
      - pydantic==2.10.6
      - pydantic-core==2.27.2
      - pyparsing==3.2.1
      - pysocks==1.7.1
      - python-dateutil==2.9.0.post0
      - pytz==2025.1
      - pyyaml==6.0.2
      - regex==2024.11.6
      - requests==2.32.3
      - sacrebleu==2.5.1
      - safetensors==0.5.3
      - scikit-image==0.24.0
      - scikit-learn==1.6.1
      - scipy==1.10.1
      - sentry-sdk==2.24.0
      - setproctitle==1.3.5
      - shapely==2.0.7
      - six==1.17.0
      - smart-open==6.4.0
      - smmap==5.0.2
      - soupsieve==2.6
      - spacy==3.7.2
      - spacy-legacy==3.0.12
      - spacy-loggers==1.0.5
      - srsly==2.5.1
      - sympy==1.13.1
      - tabulate==0.9.0
      - termcolor==2.5.0
      - thinc==8.2.5
      - threadpoolctl==3.6.0
      - tifffile==2025.3.13
      - timm==1.0.7
      - tokenizers==0.19.1
      - torch==2.3.1+cu118
      - torchvision==0.18.1+cu118
      - tqdm==4.67.1
      - transformers==4.42.3
      - triton==2.3.1
      - typer==0.9.4
      - typing-extensions==4.12.2
      - tzdata==2025.1
      - urllib3==2.3.0
      - wandb==0.19.8
      - wasabi==1.1.3
      - weasel==0.3.4
      - yacs==0.1.8

**Comment by ga1i13o** [comment]
I guess it's definitely not a GPU memory problem 😅 nice setup!
The environment looks ok, except it seems you have installed cuda-toolkit via conda-forge, which should ok, but it's the only difference wrt mine. Please try to recreate the env installing pytorch via pip rather than conda, with the command listed in the README:
`pip install torch==2.3.1 torchvision==0.18.1 --index-url https://download.pytorch.org/whl/cu118`
this install the torch binaries shipped with cuda already included
 then proceed by installing the packages listed in the requirements

moreover, does the inference code work? or is it only the training that gives segmentation fault

**Comment by ksk2023** [comment]
Dear author，

Thank you for your guidance! My code is now running successfully. Here's what I did:

Initially, my Linux system had CUDA version 12.4 installed. In the virtual environment, I installed CUDA Toolkit version 11.8 using the command `conda install cudatoolkit=11.8`.

After your guidance, I took the following steps:
I suspected that the system - level CUDA version (the one shown by `nvcc -V`) was too high. So, I re - configured the system - level CUDA to version 11.8. Here is the output:
```
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005 - 2022 NVIDIA Corporation
Built on Wed_Sep_21_10:33:58_PDT_2022
Cuda compilation tools, release 11.8, V11.8.89
Build cuda_11.8.r11.8/compiler.31833905_0
```
Then, I removed the CUDA Toolkit installed via Conda from my virtual environment. After that, I executed the code:
```
python3 main.py --resume=pretrain/pretrained_model.pth --dataset_file ytvos --HSA --use_cme_head --name_exp first --epochs 4 --batch_size 1
```
And the model ran successfully!

Out of curiosity, I reinstalled the CUDA Toolkit version 11.8 via Conda, and to my surprise, the code still ran. Here is a partial output of the results:
```
Epoch: [0]  [     0/342544]  eta: 5 days, 14:19:07  lr: 0.000010  loss: 0.0461 (0.0461)  loss_mask: 0.0053 (0.0053)  loss_dice: 0.0408 (0.0408)  loss_mask_unscaled: 0.0053 (0.0053)  loss_dice_unscaled: 0.0408 (0.0408)  grad_norm: 0.3525 (0.3525)  time: 1.4116  data: 0.3393  max mem: 3370
```

Thank you so much for your help. Although my code is running smoothly now, I have a question regarding the command you provided (`pip install torch==2.3.1 torchvision==0.18.1 --index-url https://download.pytorch.org/whl/cu118`). It seems that it didn't download the CUDA Toolkit. However, it did download some related libraries:
```
# Name                    Version                   Build  Channel
nvidia-cuda-cupti-cu11    11.8.87                  pypi_0    pypi
nvidia-cuda-nvrtc-cu11    11.8.89                  pypi_0    pypi
nvidia-cuda-runtime-cu11  11.8.89                  pypi_0    pypi
```
The command also downloaded cuDNN, as shown in the terminal output:
```
# Name                    Version                   Build  Channel
nvidia-cudnn-cu11         8.7.0.84                 pypi_0    pypi
```

I'd really like to discuss this with you. Are the CUDA components in your statement "this install the torch binaries shipped with cuda already included" referring to `nvidia-cuda-runtime-cu11`, `nvidia-cuda-nvrtc-cu11`, and `nvidia-cuda-cupti-cu11`? And do I need to download the CUDA Toolkit via Conda when creating virtual environments in the future?

I'm looking forward to your reply. Once again, thank you so much for your great help!

Best regards,
ksk2023
